src/test/java/com/zendesk/maxwell/MysqlIsolatedServer.java

```diff
@@ -32,6 +32,10 @@ public void boot(String xtraParams) throws IOException, SQLException, Interrupte
 		if ( xtraParams == null )
 			xtraParams = "";
 
+		// By default, MySQL doesn't run under root. However, in an environment like Docker, this is a reality.
+		// By checking the HOME env variable for "/root", we can identify whether we are the root user
+		boolean isRoot = System.getenv().get("HOME").equals("/root");
```

Comment:
I agree it's a little shaky. Another option is to do System.getProperty("user.name").equals("root") here. I added it just now.

I wouldn't add an env variable to Docker just for testing. I though about doing it first but then I felt stronger keeping the Docker image clean (as it's being used for production) and making sure the tests just run regardless of it's environment (there is no security issue here I guess). It also keeps the dependency on the root user local and visible to anyone editing that class.